experiments/profile_join.py

```
@@ -0,0 +1,71 @@
+    cat = Counter()
+    for off, c in counts.items():
+        i = by_off.get(off)
+        if i is None:
+            cat["<unknown pc>"] += c
+            continue
+        op = i.op
+        if op in LOADS_IND and i.args.get("base") == SP:
+            cat["spill load (SP)"] += c
+        elif op in STORES_IND and i.args.get("base") == SP:
+            cat["spill store (SP)"] += c
+        elif op in LOADS_IND or op.startswith("Load") and op not in ("LoadImm", "LoadImm64", "LoadImmJump", "LoadImmJumpInd"):
```

Ruff (0.15.15)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)